### PR TITLE
extend exception for invalid `offset_name`

### PR DESCRIPTION
**Pull request by Borda**
*Issue #, if available:* it is quite hard to understand why the exception was raised when you do not see the values

*Description of changes:* extend the exception, which will help a user understand what went wrong...

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Please tag this pr with at least one of these labels to make our release process faster:** BREAKING, new feature, bug fix, other change, dev setup

cc: @jaheba, @lostella, @kashif, @rshyamsundar

### CODE REVIEW

**Comment by kashif** [comment]
lets fix the issue, which is i believe that pandas has deprecated some of the freq strings. 
Screenshot 2024-01-29 at 11 44 14
Screenshot 2024-01-29 at 11 46 01

**Comment by Borda** [comment]
> lets fix the issue, which is i believe that pandas has deprecated some of the freq strings.

That is a good point; if you do not mind, I will split it into two PRs as these springs are used in multiple places, and this PR keeps focused on the exception enrichment... So, I would take the conversion of the string just after :)

**Comment by Borda** [comment]
@kashif, based on your reaction, if you are fine with keeping this PR simple, would you mind approving Ci and reviewing it? :)

**Comment by kashif** [comment]
I am not a maintainer so have no such power... but I believe they will review it

File: src/gluonts/time_feature/lag.py
Comment (lostella) [inline src/gluonts/time_feature/lag.py:149]:
```suggestion
        raise ValueError(f"invalid frequency: {freq_str=}, {offset_name=}")
```

**Comment by lostella** [comment]
@Borda thanks for this! Just left a comment on improving the message, let me know what you think.

There’s a lot of red in the workflows, I’ll take a look and try to unblock PRs

**Review by lostella — APPROVED** [review]

**Comment by Borda** [comment]
> thanks for this! Just left a comment on improving the message, let me know what you think.

yes, that looks better, but seems you merged it before I could update it

> There’s a lot of red in the workflows, I’ll take a look and try to unblock PRs

**Pull request merged**
